# 目标物体圆周运动轨迹配置
# 空间单位为米，时间单位为秒，角度单位为弧度

/**:
  ros__parameters:
    # 状态机交互参数
    drone_id: 0                  # 无人机ID，用于订阅状态话题
    
    # 圆周运动参数
    circle_radius: 1.5          # 圆形轨迹半径 [m] 必须是浮点数！！！
    circle_center_x: 0.0        # 圆心 X 坐标 (NED坐标系) [m]
    circle_center_y: 0.0        # 圆心 Y 坐标 (NED坐标系) [m]
    circle_center_z: -1.2       # 圆心 Z 坐标 (NED坐标系，负值表示高度) [m]
    circle_init_phase: 3.1415926   # 初始相位角 [rad] (π/2，使目标初始化在(0, 2))
    circle_times: 1             # 圆周运动圈数
    circle_duration: 20.0       # 单圈持续时间 [s] - r=2时对应线速度约0.628 m/s r=1.5时对应0.471 m/s
    # circle_duration: 12.6      # 单圈持续时间 [s] - r=2时对应线速度1m/s
    
    # 运动平滑参数
    ramp_up_time: 3.0           # 加速时间 [s]
    ramp_down_time: 3.0         # 减速时间 [s]
    stationary_time: 3.0        # 初始静止时间 [s]（在圆周运动开始前保持静止的时间）
    
    # 控制参数
    timer_period: 0.02          # 发布周期 (50Hz) [s]
    max_speed: -1.0             # 最大线速度限制 [m/s] (负值表示不限制)

